src/main/asciidoc/reference/vault-secret-engines.adoc

```diff
@@ -0,0 +1,252 @@
+The input `String` uses the default `Charset` to encode the string into its binary representation.
+Requesting a token returns a `VaultToken` which is used as value object for Vault tokens.
+The `encrypt` method returns Base64-encoded ciphertext, typically starting with `vault:`.
+<3> To decrypt ciphertext into plaintext, call the `decrypt` method.
+It decrypts the ciphertext and returns a `String` that is decoded using the default charset.
+====
+
+The preceeding example uses simple strings for cryptographic operations.
+While it's a simple approach, it bears the risk of charset misconfiguration and it is not binary-safe.
+Binary safety is required when the plaintext uses a binary representation for data such as images, compressed data or binary datastructures.
+
+To encrypt and decrypt binary data, use the `Plaintext` and `Ciphertext` value objects that can hold binary values:
+
+====
+[source,java,indent=0]
+----
+include::../{example-root}/Transit.java[tags=encryptPlaintext]
+----
+<1> Assuming a key `my-aes-key` is already in place, we're encrypting the `Plaintext` object.
+In return, the `encrypt` method returns a `Ciphertext` object.
+<2> The `Ciphertext` object can be used directly for decryption and returns a `Plaintext` object.
+====
+
+`Plaintext` and `Ciphertext` come with a contextual object `VaultTransitContext`.
+It is used to supply a nonce value for https://www.vaultproject.io/docs/secrets/transit#convergent-encryption[convergent encryption] and for a context value to make use of key derivation.
+
+Transit allows for signing plaintext and verifying the signature for a given plaintext.
+Sign operations require an asymmetric key, typically using Elliptic Curve Cryptography or RSA.
+
+TIP: This tip explains how signatures work internally. +
+ +
+Signatures make use of the public/private key split to ensure authenticity. +
+The signer uses its private key to create a signature (as otherwise anybody would be able to sign messages in your name!).
+The verifier uses the public key part to verify the signature. The actual signature is typically a hash value. +
+ +
+Internally, the hash gets computed and encrypted using the private key to create the final signature. The verification decrypts the signature message, computes their own hash for the plaintext and compares both hash values to check whether the signature is valid or not.
+
+====
+[source,java,indent=0]
+----
+include::../{example-root}/Transit.java[tags=signVerify]
+----
+<1> Signing requires an asymmetric key. Any Elliptic Curve Cryptography or RSA key type can be used. Once the key is created, we have all pre-requisites in place to create a signature.
+<2> The signature gets created for a plaintext message. The returned `Signature` contains an ASCII-safe string that uses Base64 characters.
+<3> To verify the signature, the verification requires a `Signature` object and the plaintext message. As return you get whether the signature was valid or not.
```

Comment:
Change to "<3> To verify the signature, the verification requires a `Signature` object and the plain text message. As the return value, you get whether the signature was valid or not."

(space in plain text and additional words and a comma around "return")